### PR TITLE
[14.0][ADD] stock_service_level

### PR DESCRIPTION
**Pull request by petrus-v**
This is draft just exposing my plan in the description.rst ! Please Feel free to comment suggest other solutions, name or repo to host those sources or any other comments.

### CODE REVIEW

File: stock_service_level/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  level.
+
+
+The module integrates seamlessly with Odoo's stock management system.
+Users can configure service levels and associated rules through the
+interface, enabling efficient management of product variants and service
+offerings within the logistics framework.
+
+
+Implementation roadmap in different modues:
```
Comment (alexandregaldeano) [inline stock_service_level/readme/DESCRIPTION.rst]:
```suggestion
Implementation roadmap in different modules:
```

File: stock_service_level/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Module `stock_service_level` in oca/wms:
+
+* [ ] Creating `stock.service.level` model
+* [ ] link stock.move to service level model
```
Comment (alexandregaldeano) [inline stock_service_level/readme/DESCRIPTION.rst]:
```suggestion
* [ ] link `stock.move` to service level model
```

**Comment by petrus-v** [comment]
note to myself to investigate:

Why gh actions appears to be testing with an old unmaintained python version (is it the current debian version when odoo 14.0 was released first)... this make unavailable some new features like walrus operators which are currently supported by the oldest supported python version: 3.8 !!

As far I remember I've used it on an other repository without error, maybe a simple copier update would fix that !

File: stock_service_level/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Foodles (http://www.foodles.co).
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock service level",
+    "summary": "Manage stock route by service level",
+    "version": "14.0.1.3.1",
```
Comment (barkat-matthias) [inline stock_service_level/__manifest__.py]: Memo to change the version to 14.0.1.0.0

**Review by barkat-matthias — APPROVED** [review]

File: stock_service_level/models/stock_service_level.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    _description = "Stock Service level"
+
+    name = fields.Char(required=True, translate=True)
+    code = fields.Char(required=True)
+
+    _sql_constraints = [
+        (
+            "book_format_code_uniq",
+            "unique(code)",
+            "Code must be uniq.",
```
Comment (rousseldenis) [inline stock_service_level/models/stock_service_level.py]:
```suggestion
            "Code must be unique.",
```

prefer explicit wording

File: stock_service_level/models/stock_move.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2024 Foodles (https://www.foodles.co)
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    service_level_id = fields.Many2one(
```
Comment (rousseldenis) [inline stock_service_level/models/stock_move.py:10]:
```suggestion
    service_level_id = fields.Many2one(
        index="btree_not_null",
```

File: stock_service_level/models/stock_service_level.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+from odoo import fields, models
+
+
+class StockServiceLevel(models.Model):
+    _name = "stock.service.level"
+    _description = "Stock Service level"
+
+    name = fields.Char(required=True, translate=True)
+    code = fields.Char(required=True)
```
Comment (rousseldenis) [inline stock_service_level/models/stock_service_level.py]:
```suggestion
    code = fields.Char(required=True, index=True)
```

**Review by rousseldenis — APPROVED** [review]
Code review. Minor comments

**Comment by petrus-v** [comment]
I declare myself as maintainer but this is a new module... sorry for this try!

**Comment by petrus-v** [comment]
Hi @rousseldenis is there any thing missing before merging here ?

Thanks for your time

**Comment by rousseldenis** [comment]
@petrus-v Could you squash your commits or improve the second one's message ?

**Comment by petrus-v** [comment]
Thanks @rousseldenis,  commits squashed ! CI is Green :fireworks:

**Comment by rousseldenis** [comment]
Comments have been attended.

/ocabot merge patch

**Pull request merged**